src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -63,6 +65,38 @@ function CreateCourse() {
 			setCourseAuthors([]);
 		}
 	};
+	const cancel = () => {
+		setView(true);
+	};
+
+	const AuthorList = ({
+		authors,
+		buttonInfo,
+		courseAuthors,
+		setCourseAuthors,
+		availableAuthors,
+		setAvailableAuthors,
+	}) => {
+		return (
+			<ul className='create-course__list'>
+				{authors.length === 0 ? (
+					<li className='create-course__list--empty'>Author list is empty</li>
+				) : (
+					authors.map((author) => (
+						<AuthorTile
+							key={author.id}
+							author={author}
+							buttonInfo={buttonInfo}
+							courseAuthors={courseAuthors}
+							setCourseAuthors={setCourseAuthors}
+							availableAuthors={availableAuthors}
+							setAvailableAuthors={setAvailableAuthors}
+						/>
+					))
+				)}
+			</ul>
+		);
+	};
```

Comment:
Here you did not exactly what I meant.

If you look at your initial code you will see that the only difference between the two pars is the array from which you are gathering data `availableAuthors`; `courseAuthors` and button text `buttonInfo`.

So those are the only thing you have to change.

```suggestion
	const renderAuthorList = (authors, buttonInfo ) => (
			<ul className='create-course__list'>
				{authors.length === 0 ? (
					<li className='create-course__list--empty'>Author list is empty</li>
				) : (
					authors.map((author) => (
						<AuthorTile
							key={author.id}
							author={author}
							buttonInfo={buttonInfo}
							courseAuthors={courseAuthors}
							setCourseAuthors={setCourseAuthors}
							availableAuthors={availableAuthors}
							setAvailableAuthors={setAvailableAuthors}
						/>
					))
				)}
			</ul>
		);
```

and lower in the code you will have something like this:

`{renderAuthorList(availableAuthors, ADD_AUTHOR)}`
and
`{renderAuthorList(courseAuthors, DELETE_AUTHOR)}`

Reply:
I have made necessary correction. Thanks a lot for your support!